src/main/java/Ui.java

```diff
@@ -0,0 +1,124 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Ui {
+    /**
+     * Starts the user interface.
+     * @throws IOException Thrown when I/O error occurs
+     */
+    public static void run() throws IOException {
+        TaskList tasks = new TaskList();
+
+        Storage.readFile(tasks);
+
+        printHorizontalLine();
+        printIndented("Hello! I'm Duke");
+        printIndented("What can I do for you?");
+        printHorizontalLine();
+        System.out.println();
+
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        Parser parse = new Parser(input, false);
+
+        while (parse.getCommand() != Command.BYE) {
```

Comment:
I feel that this program logic should not belong in the UI but in Duke or another separate module. The UI should only handle input and output.